h2o-algos/src/main/java/hex/glm/GLMUtils.java

```diff
@@ -31,6 +33,52 @@
     return gamColIndices;
   }
 
+  public static TwoDimTable combineScoringHistory(TwoDimTable glmSc1, TwoDimTable earlyStopSc2) {
+    String[] esColTypes = earlyStopSc2.getColTypes();
+    String[] esColHeaders = earlyStopSc2.getRowHeaders();
+    String[] esColFormats = earlyStopSc2.getColFormats();
+    ArrayList<String> finalRowHeaders = new ArrayList<>(Arrays.asList(glmSc1.getRowHeaders()));
```

Comment:
I think that interface List should be used instead of the ArrayList for variable type definition on all places (I did not notice usage of anything ArrayList specific in the code) - then, you don't have to use the ArrayList constructor and remove redundant creation of another list and just assign the return value of Arrays.asList to variable

Reply:
@koniecsveta:  I am trying to combine two lists into one final list.  Since I do not know how long the final list is going to be, I use ArrayList which will automatically expand to accommodate any final length I want.  In addition, I also need to preserve the order of the combination of the two lists as well.

Reply:
I agree with the ArrayList usage but the comment was suggesting this change (for all five list initializations):

`List<String> finalRowHeaders = Arrays.asList(glmSc1.getRowHeaders());`

since the interface should be used for variable type definition and Arrays.asList already creates new instance of ArrayList, therefore, putting this as constructor parameter is unnecessary

Reply:
what zuzana is saying is that this should be `List<String> finalRowHeaders = ...`

Reply:
Actually, I need to do List<String> finalRowHeaders =new ArrayList<>( Arrays.asList(glmSc1.getRowHeaders()));

in order to be able to use finalRowHeaders.add(colName);

Reply:
java.util.Arrays.asList() looks like this:

`public static <T> List<T> asList(T... a) {
        return new ArrayList<>(a);
    }`

so by doing
` new ArrayList<>( Arrays.asList(glmSc1.getRowHeaders()));`
you actually do 
`new ArrayList<>(new ArrayList<>(glmSc1.getRowHeaders()))`

Reply:
Got it.  I think I got them all.